beacon-chain/rpc/eth/rewards/service.go

```diff
@@ -124,6 +127,22 @@ func (rs *BlockRewardService) GetStateForRewards(ctx context.Context, blk interf
 	// We want to run several block processing functions that update the proposer's balance.
 	// This will allow us to calculate proposer rewards for each operation (atts, slashings etc).
 	// To do this, we replay the state up to the block's slot, but before processing the block.
+
+	// Try getting the state from the next slot cache first.
```

Comment:
I think this is meant to be "previous" since we are getting the previous roots here.
```suggestion
	// Try getting the state from the previous slot cache first.
```

Reply:
No, we are using the `nextSlotCache` here - it caches the state for the _next_ slot